src/plugins/intel_gpu/tests/unit/test_cases/gemm_gpu_test.cpp

```diff
@@ -1195,21 +1195,19 @@ struct gemm_base_test_params {
 #define CASE_GEMM_FP16_TILED_NN_BROADCAST_4 8, 8, 8, 1, 1, 2, 2, 2, 2, 2, 2, false, false, \
 1.0f, 4.0f, data_types::f16, data_types::f16, data_types::f16, data_types::f16, { -1, 1, 1 }, { -1, 1, 1 }, { -1, 1, 1 }
 
-#endif // ENABLE_ONEDNN_FOR_GPU
-
 template <typename gemm_params, typename input0_type, typename input1_type, typename input2_type, typename output_type, typename accumulator_type>
 class GemmBaseTest : public ::testing::TestWithParam<gemm_params> {
 public:
+    virtual ov::intel_gpu::ImplementationDesc getImplementationDesc(gemm_params& p) {
+         return { format::bfyx, p.kernel_name };
+    }
 
     inline size_t getGemmIndex(size_t x, size_t y, size_t f, size_t b, size_t x_size, size_t y_size, size_t f_num, size_t b_num,
                                size_t x_pitch, size_t y_pitch, size_t f_pitch, size_t b_pitch) {
         return (x % x_size) * x_pitch + (y % y_size) * y_pitch + (f % f_num) * f_pitch + (b % b_num) * b_pitch;
     }
 
     void execute(gemm_params& p, bool is_caching_test = false) {
-        auto& engine = get_test_engine();
-        if (!engine.get_device_info().supports_immad)
-            return;
         auto y0_size = p.m_size;
```

Comment:
is this a common test code for iGPU and dGPU?

Reply:
Yes, iGPU directly uses GemmBaseTest, dGPU uses GemmBaseOneDNNTest which inherits GemmBaseTest.